lib/Protocol/FIX/Field.pm

```diff
@@ -42,15 +42,15 @@ The following field types are known to the class. Validators are provided.
 =cut
 
 # anyting defined and not containing delimiter
-my $BOOLEAN_validator  = sub { defined($_[0]) && $_[0] =~ /^[YN]$/ };
-my $STRING_validator   = sub { defined($_[0]) && $_[0] !~ /$Protocol::FIX::TAG_SEPARATOR/ };
-my $INT_validator      = sub { defined($_[0]) && $_[0] =~ /^-?\d+$/ };
-my $LENGTH_validator   = sub { defined($_[0]) && $_[0] =~ /^\d+$/ && $_[0] > 0 };
+my $BOOLEAN_validator  = sub { defined($_[0]) && $_[0] =~ /^[YN]$/o };
```

Comment:
Never use `/o`, it's useless for performance since about 5.12 or so, and it can be actively harmful if there's any interpolation involved.

The documentation literally states this in recent versions of Perl:

```
o  - pretend to optimize your code, but actually introduce bugs
```

Reply:
I saw minor improvement in `nytprof`. Will try to substitute to `qr//`